### PR TITLE
Add checks for cache dir validity

### PR DESCRIPTION
**Pull request by daemontus**
Wasn't sure where to put the unit tests, so I created a new class for them :)

### CODE REVIEW

File: runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.File;
+import java.security.InvalidParameterException;
+
+/**
+ * Created by daemontus on 02/09/16.
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class InvalidUsageTest {
+    @ClassRule
+    public static TemporaryFolder temporaryFolder = new TemporaryFolder();
```
Comment (VictorAlbertos) [inline runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java]: Remove states modifier

File: runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.io.File;
+import java.security.InvalidParameterException;
+
+/**
+ * Created by daemontus on 02/09/16.
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class InvalidUsageTest {
+    @ClassRule
```
Comment (VictorAlbertos) [inline runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java]: Remove @ClassRule annotation

File: runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import org.junit.rules.TemporaryFolder;
+import org.junit.runners.MethodSorters;
+
+import java.io.File;
+import java.security.InvalidParameterException;
+
+/**
+ * Created by daemontus on 02/09/16.
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
```
Comment (VictorAlbertos) [inline runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java]: Remove @FixMethodOrder annotation

File: runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+public class InvalidUsageTest {
+    @ClassRule
+    public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test(expected = InvalidParameterException.class)
+    public void _00_Cache_Directory_Null() {
```
Comment (VictorAlbertos) [inline runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java]: Remove _00_ from the name method. We don't need that the methods preserve any order when they are executed.

File: runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        new RxCache.Builder()
+            .persistence(null, Jolyglot$.newInstance());
+    }
+
+    @Test(expected = InvalidParameterException.class)
+    public void _01_Jolyglot_Null() {
```
Comment (VictorAlbertos) [inline runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java]: Remove _01_ from the name method. We don't need that the methods preserve any order when they are executed.

Comment (VictorAlbertos) [inline runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java]: Remove _00_ from the name method. We don't need that the methods preserve any order when they are executed.

File: runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        new RxCache.Builder()
+                .persistence(temporaryFolder.getRoot(), null);
+    }
+
+    @Test(expected = InvalidParameterException.class)
+    public void _00_Cache_Directory_Not_Exist() {
```
Comment (VictorAlbertos) [inline runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java]: Remove _00_ from the name method. We don't need that the methods preserve any order when they are executed.

Comment (VictorAlbertos) [inline runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java]: Remove _00_ from the name method. We don't need that the methods preserve any order when they are executed.

File: runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import org.junit.runners.MethodSorters;
+
+import java.io.File;
+import java.security.InvalidParameterException;
+
+/**
+ * Created by daemontus on 02/09/16.
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class InvalidUsageTest {
```
Comment (VictorAlbertos) [inline runtime/src/test/java/io/rx_cache/internal/InvalidUsageTest.java]: Rename to RxCacheBuilderValidationTest

**Comment by VictorAlbertos** [comment]
Thanks :)

I've added some comments to improve the test a little bit. Do you mind fixing them and commit the changes to this same PR?

Basically, we don't need that the reference to the folder be static, for the same reason that we don't need that the test are executed in any specific order, that's it because we don't need to persevere any state of the disk between tests (as opposed to other test where we really need that because we are doing integration testing more than unit testing. But this one should be truly unit testing).

**Comment by daemontus** [comment]
Sure, good point, here you go.

**Comment by daemontus** [comment]
Well this is weird - local check passed, but travis reports some sort of dead/livelock. Any ideas about this?

**Comment by VictorAlbertos** [comment]
I have no idea xD But don't worry, it happens from time.

I'm gonna to merge it right now.

Thanks again ;)

**Pull request merged**